core/src/main/java/net/sourceforge/jnlp/runtime/classloader/DelegatingClassLoader.java

```diff
@@ -0,0 +1,73 @@
+package net.sourceforge.jnlp.runtime.classloader;
+
+import java.net.URL;
+import java.util.Objects;
+
+public class DelegatingClassLoader extends ClassLoader
+{
+    public static final DelegatingClassLoader instance = new DelegatingClassLoader(Thread.currentThread().getContextClassLoader());
+    private ClassLoader classLoader;
+
+    public static DelegatingClassLoader getInstance()
+    {
+        return instance;
+    }
+
+    DelegatingClassLoader(ClassLoader loader)
+    {
+        super(loader);
+        this.classLoader = loader;
+    }
+
+    public void setClassLoader(ClassLoader loader)
+    {
+        this.classLoader = loader;
+    }
+
+    protected Class findClass(String name) throws ClassNotFoundException
+    {
+        return this.classLoader.loadClass(name);
+    }
+
+    protected URL findResource(String name)
+    {
+        return this.classLoader.getResource(name);
+    }
+
+    public int hashCode()
+    {
+        int result = 1;
+        result = 31 * result + (this.classLoader == null ? 0 : this.classLoader.hashCode());
+        result = 31 * result + (this.getParent() == null ? 0 : this.getParent().hashCode());
+        return result;
+    }
+
+    public boolean equals(Object obj)
+    {
+        if (this == obj)
+        {
+            return true;
+        }
+        else if (obj == null)
+        {
+            return false;
+        }
+        else if (this.getClass() != obj.getClass())
+        {
+            return false;
+        }
+        else
+        {
+            DelegatingClassLoader other = (DelegatingClassLoader) obj;
+            if(Objects.equals(this.classLoader,other.classLoader)==false)
+            {
+                return false;
+            }
+            if(Objects.equals(this.getParent(),other.getParent()) == false)
```

Comment:
```suggestion
            if(!Objects.equals(this.getParent(),other.getParent()))
```

Reply:
This is a personal thing :) I really dont like the ! because it is so tiny and quickly overlooked. But im perfectly ok if you want to have it that way

Reply:
fixed

Reply:
fine with it but in ITW / OWS we normally use the '!'. Thanks for fixing it.